File: 백준/Silver/11723. 집합/집합.py
import sys
input = sys.stdin.readline
n = int(input())
def sol(order, s):
    if order == "all":
        tmp = [i for i in range(1, 20+1)]
        s = set(tmp)
    elif order == "empty":
        s = set([])
    else:
        odr, num = map(str, order.split())
        if odr == "add" and int(num) not in s:
            s.add(int(num))
        elif odr == "remove":
            if len(s) > 0 and int(num) in s:
                s.remove(int(num))
        elif odr == "check":
            if int(num) in s:
                print(1)
            else:
                print(0)
        else:
            if int(num) in s:
                s.remove(int(num))
            else:
                s.add(int(num))
    return s
s = set([])
for i in range(n):
    order = input().strip()
    s = sol(order, s)

File: 백준/Silver/11723. 집합/집합.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws Exception {	
		StringBuilder sb = new StringBuilder();
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		int M = Integer.parseInt(br.readLine());
		int bit = 0;
		while(M-->0) {
			st = new StringTokenizer(br.readLine());
			String s = st.nextToken();
			if("all".equals(s)) {
				bit = (1 << 21) - 1;
			} else if("empty".equals(s)) {
				bit = 0;
			} else {
				int x = Integer.parseInt(st.nextToken());
				if("add".equals(s)) {
					bit |= 1 << x;
				} else if("remove".equals(s)) {
					bit &= ~(1 << x);
				} else if("check".equals(s)) {
					sb.append((bit & (1 << x)) == 0 ? 0 : 1).append('\n');
				} else if("toggle".equals(s)) {
					bit ^= 1 << x;
				} 
			}
		}
		System.out.println(sb);
	}

	public static int readInt() throws Exception {
		int val = 0;
		int c = System.in.read();
		while (c <= ' ') {
			c = System.in.read();
		}
		boolean flag = (c == '-');
		if (flag)
			c = System.in.read();
		do {
			val = 10 * val + c - 48;
		} while ((c = System.in.read()) >= 48 && c <= 57);

		if (flag)
			return -val;
		return val;
	}
}